src/components/cc-pricing-product-consumption/cc-pricing-product-consumption.js

```diff
@@ -268,142 +328,124 @@ export class CcPricingProductConsumption extends withResizeObserver(LitElement)
         };
       });
 
-      this._simulator = new PricingConsumptionSimulator(this.sections);
+      this._simulator = new PricingConsumptionSimulator(this.product.sectionList);
     }
   }
 
   render () {
+    return html`
+      ${this.product.state === 'error' ? html`
+        <cc-notice intent="danger" message=${i18n('cc-pricing-product-consumption.error')}></cc-notice>
+      ` : ''}
 
-    const someNullishIntervals = this.sections
-      ?.some(({ intervals }) => intervals == null) ?? true;
-    const everyQuantityAtZero = this.sections
-      ?.map(({ type }) => this._simulator.getQuantity(type))
-      ?.every((quantity) => quantity === 0) ?? true;
+      ${this.product.state === 'loading' ? html`
+        <cc-loader></cc-loader>
+      ` : ''}
 
-    const name = this.name ?? SKELETON_NAME;
+      ${this.product.state === 'loaded' ? this._renderLoaded() : ''}
```

Comment:
suggestion: I would pass the `this.product.sectionList` to `_renderLoaded()` with the type `PricingSection[]` to help typechecking